### PR TITLE
feat(adapter): add collection.xyz TVL adapter

### PR DESCRIPTION
**Pull request by phattv**
**NOTE**

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. The protocol is usually listed within 24 hours of merging the PR
4. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
5. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
6. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
7. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):

Collection.xyz

##### Twitter Link:

https://twitter.com/CollectionWeb3

##### List of audit links if any:

https://github.com/collectionswap/collectionswap/blob/main/audits/ABDK_Collection_CollectionSwap_v_1_0_1_.pdf

##### Website Link:

https://collection.xyz/

##### Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders):

https://collection.xyz/logo.svg

##### Current TVL:

804.75 k

##### Chain:

Ethereum

##### Short Description (to be shown on DefiLlama):

> Collection.xyz NFT DEX AMM | Automatically buy, sell & earn yield on your NFTs | Pool NFTs based on token IDs, traits or rarity | Composable with DeFi / NFTfi protocols, enjoy 0% protocol fees, live on Ethereum Mainnet.

##### Category (full list at https://defillama.com/categories) *Please choose only one:

NFT Marketplace

##### methodology (what is being counted as tvl, how is tvl being calculated):

Summing up the bid and ask liquidity within every pool created by the deployed factory contract

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/collectionxyz exports TVL: 
        
 
 
    ethereum                  26.13 M
    
    total                    26.13 M

**Comment by phattv** [comment]
> The adapter at projects/collectionxyz exports TVL:
> 
> ```
> ethereum                  26.13 M
> 
> total                    26.13 M 
> ```

@llamatester how can have a break down of TVL in ETH for each NFT collection, i'm trying to do that
```
{
  ethereum: 380.50568972608147,
  '0xb7f7f6c52f2e2fdb1963eab30438024864c313f6': 115.14,
  '0xbc4ca0eda7647a8ab7c2061c2e118a18a936f13d': 76.0815,
  '0x9c8ff314c9bc7f6e59a9d9225fb22946427edc03': 50,
  ...
}
```

if i only return `{ethereum: 380.50568972608147}` it works as expected

**Comment by llamatester** [comment]
The adapter at projects/collectionxyz exports TVL: 
        
 
 
    ethereum                  762.08 k
    
    total                    762.08 k

**Comment by phattv** [comment]
updated the PR request, there are some real time changes to the TVL

**Comment by g1nt0ki** [comment]
@phattv thanks for the PR, I have refactored it here to be on chain, tvl is lower because certain collections in your tvl are not whitelisted/we dont have prices for, can you review my changes? https://github.com/DefiLlama/DefiLlama-Adapters/pull/5925

**Comment by phattv** [comment]
> @phattv thanks for the PR, I have refactored it here to be on chain, tvl is lower because certain collections in your tvl are not whitelisted/we dont have prices for, can you review my changes? #5925

@g1nt0ki yes please use yours, thanks very much for the help, shall i close this MR?
